Automattic/wp-calypso
# Store: Update product category selectors & tests, run prettier

This PR does some cleanup on the product categories components & selectors, including running prettier, sorting selectors alphabetically, and moving the product category state into a fixtures file.

The biggest change comes in the selectors:

- `areProductCategoriesLoadingIgnoringPage` has been renamed to `areAnyProductCategoriesLoading`
- `getProductCategoriesIgnoringPage` has been renamed to `getAllProductCategories`
- Added an internal-use `getRawCategoryState` helper so that we don't need to call the entire state path each time
- Switched `getProductCategories` to use `getProductCategory` for fetching each category
- Renamed `rootState` in each function to `state` to match other Calypso selectors

I've also run prettier on the product-categories list & applies-to-filtered-list. This is in preparation for another PR, so that all the "cleanup" changes live in one PR.

**To test**

- Check that there are no UI changes, specifically on
  - product categories on promotions
  - product category list view
  - product categories on product create/edit pages
- Make sure the tests pass: `npm run test-client client/extensions/woocommerce/state/sites/product-categories`
